pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -247,20 +232,27 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			if isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl)) {
+			if (isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl))) &&
 				// if the workload is Admitted we don't want to retry on BookingExpired
+				ac != nil && ac.State == kueue.CheckStatePending {
+				// if the workload is in Retry/Rejected state we don't create another ProvReq
 				attempt = getAttempt(log, oldPr, wl.Name, checkName)
-				if attempt <= c.maxRetries {
-					remainingTime := c.remainingTimeToRetry(oldPr, attempt)
-					if remainingTime <= 0 {
-						shouldCreatePr = true
-						attempt += 1
-					} else if requeAfter == nil || remainingTime < *requeAfter {
-						requeAfter = &remainingTime
-					}
+				if !features.Enabled(features.KeepQuotaForProvReqRetry) {
+					shouldCreatePr = true
+					attempt += 1
+				}
+				remainingTime := c.remainingTimeToRetry(oldPr, attempt, prc)
+				if remainingTime <= 0 {
+					shouldCreatePr = true
+					attempt += 1
```

Comment:
This doesn't look right, if KeepQuotaForProvReqRetry disabled, and remainingTime=0, then we will bump `attempt` by 2.

Reply:
You're right, fixed